test/transformer/descriptor/generic/extends.test.ts

```diff
@@ -102,4 +103,145 @@ describe('for generic', () => {
             expect(properties.b).toBe(0);
         });
     });
+
+    describe('with nested generics that extends itself', () => {
+        interface A<T> {
+            a: T;
+            b: A<T>;
+        }
+
+        interface B<T> extends A<T> {
+            c: number;
+        }
+
+        it('should set the generic value', () => {
+            const properties: B<string> = createMock<B<string>>();
+            expect(properties.a).toBe('');
+            expect(properties.c).toBe(0);
+            expect(properties.b.b.a).toBe('');
+        });
+    });
+
+    describe('with nested generics classes that extends itself', () => {
+        class A<T> {
+            public a: T;
+            public b: A<T>;
+        }
+
+        class B<T> extends A<T> {
+            public c: number;
+            private d: T;
+        }
+
+        it('should set the generic value', () => {
+            const properties: B<number> = createMock<B<number>>();
+            expect(properties.a).toBe(0);
+            expect(properties.c).toBe(0);
+            expect(properties.b.b.a).toBe(0);
+            // @ts-ignore
+            expect(properties.d).toBeUndefined();
+        });
+    });
+
+    describe('with nested generics types that extends itself', () => {
+        // tslint:disable-next-line
+        type A<T> = {
+             a: T;
+             b: A<T>;
+        };
+
+        interface B<T> extends A<T> {
+            c: number;
+            d: T;
+        }
+
+        it('should set the generic value', () => {
+            const properties: B<number> = createMock<B<number>>();
+            expect(properties.a).toBe(0);
+            expect(properties.c).toBe(0);
+            expect(properties.b.b.a).toBe(0);
+        });
+    });
+
+    describe('with generics that extends typescript libraries', () => {
+        interface B<T> extends Array<T> {
+            c: number;
+            d: T;
+        }
+
+        it('should not consider typescript library', () => {
+            const properties: B<number> = createMock<B<number>>();
+            expect(properties.c).toBe(0);
+            expect(properties).not.toBe([]);
+        });
+    });
+
+    describe('with multiple extends', () => {
+        interface A<T> {
+            a: T;
+        }
+
+        interface C<T> {
+            c: T;
+        }
+
+        interface B<T1, T2, T3> extends A<T1>, C<T2> {
+            b: T3;
+        }
+
+        it('should not consider typescript library', () => {
```

Comment:
this should not say that, it should say something like "should set the generic value"

Reply:
thanks!